File: app/src/main/java/com/loopeer/codereader/ui/view/LoginChecker.java
package com.loopeer.codereader.ui.view;

import android.text.TextUtils;

public class LoginChecker extends Checker {

    public String username;
    public String password;

    public LoginChecker(CheckObserver checkObserver) {
        super(checkObserver);
    }

    public void setUsername(String username) {
        this.username = username;
        mCheckObserver.check(isEnable());
    }

    public void setPassword(String password) {
        this.password = password;
        mCheckObserver.check(isEnable());
    }

    @Override
    boolean isEnable() {
        return !TextUtils.isEmpty(username) && !TextUtils.isEmpty(password);
    }
}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/ui/view/LoginChecker.kt
package com.loopeer.codereaderkt.ui.view

import android.text.TextUtils

class LoginChecker(checkObserver: Checker.CheckObserver) : Checker(checkObserver) {

    var username: String?=null
    set(value) {
        field=value
        mCheckObserver.check(isEnable)
    }
    var password: String?=null
    set(value) {
        field=value
        mCheckObserver.check(isEnable)
    }


     override val isEnable: Boolean
        get() = !TextUtils.isEmpty(username) && !TextUtils.isEmpty(password)
}
